libshaderc_util/src/spirv_tools_wrapper.cc

```diff
@@ -79,4 +82,41 @@ bool SpirvToolsAssemble(const string_piece assembly, spv_binary* binary,
   return result;
 }
 
+bool SpirvToolsOptimize(const std::vector<PassId>& enabled_passes,
+                        std::vector<uint32_t>* binary, std::string* errors) {
+  errors->clear();
+  if (enabled_passes.empty()) return true;
+  if (std::all_of(
+          enabled_passes.cbegin(), enabled_passes.cend(),
+          [](const PassId& pass) { return pass == PassId::kNullPass; })) {
+    return true;
+  }
+
+  spvtools::Optimizer optimizer(SPV_ENV_VULKAN_1_0);
```

Comment:
Hm.   This is a smell.  It seems like we should be passing in the environment or a context object.

But ok for now.